default.nix

```diff
@@ -18,11 +17,11 @@ let haskellNix = {
         )]
         else []
     ) ++ (
-      if defaultCompilerNixName != null
+      if args ? defaultCompilerNixName
         then [(
           final: prev: {
             haskell-nix = prev.haskell-nix // {
-              inherit defaultCompilerNixName;
+              userCompilerNixName = args.defaultCompilerNixName;
```

Comment:
I'd be tempted to just drop it and make this an error...

Reply:
We need something to make the tests work though.  I quite like the idea of replacing this with something like:
```
haskell-nix = prev.haskell-nix // {
  withGhc = compiler-nix-name: {
    cabalProject = args: final.haskell-nix.cabalProject ({ inherit compiler-nix-name; } // args);
    hackage-package = args: final.haskell-nix.hackage-package ({ inherit compiler-nix-name; } // args);
    ...
  };
};
```

Reply:
Why do the tests need to set `defaultNixName`? Shouldn't they all be setting explicit `compiler-nix-name`s for their individual tests?

Reply:
Yes I'm going to update all the tests so they do that instead of using `defaultCompilerNixName`.  I was hoping there was a way to avoid that, but @angerman pointed out it is nicer to be explicit here.  It makes the tests more useful as examples.

Reply:
Test suite no longer uses defaultCompilerNixName.

Reply:
I thought the point of that change was so we could then get rid of the `userCompilerNixName` thing? If we no longer need it for tests then let's just delete it.

The only other reason I can see to keep it is backwards compatibility, but I buy Rodney's argument that that's something of a losing battle anyway and we should just keep a changelog.